hydra-cluster/src/Hydra/Generator.hs

```diff
@@ -130,3 +125,59 @@ genDatasetConstantUTxO faucetSk nClients nTxs = do
     pure ClientDataset{clientKeys, initialUTxO, txSequence}
 
   thrd (_, _, c) = c
+
+-- Prepare funding transaction which will give every client's
+-- 'externalSigningKey' "some" lovelace. The internal 'signingKey' will get
+-- funded in the beginning of the benchmark run.
+makeGenesisFundingTx :: SigningKey PaymentKey -> [ClientKeys] -> Gen Tx
+makeGenesisFundingTx faucetSk clientKeys = do
+  let nClients = length clientKeys
+  clientFunds <- forM clientKeys $ \ClientKeys{externalSigningKey} -> do
+    amount <- Coin <$> choose (1, availableInitialFunds `div` fromIntegral nClients)
+    -- NOTE: It seems if this becomes choose (1,1) we get funny signingKeys;
+    -- i.e. like "0001010100010001000000010100000001010001000101000000010101010001".
+    pure (getVerificationKey externalSigningKey, amount)
+  let fundingTransaction =
+        mkGenesisTx
+          networkId
+          faucetSk
+          (Coin availableInitialFunds)
+          clientFunds
+  pure fundingTransaction
+
+genDatasetConstantUTxODemo ::
+  -- | The faucet keys
+  (VerificationKey PaymentKey, SigningKey PaymentKey) ->
```

Comment:
Why do we need to pass in the keys where we know that funds must be available at the `Faucet`?

Should: not need this.